repository_service_tuf/cli/admin/metadata.py

```diff
@@ -0,0 +1,605 @@
+) -> bool:
+    """Verify that key with `keyid` have been used to sign the current root"""
+    if keyid in already_loaded_keyids:
+        console.print(
+            ":cross_mark: [red]Failed[/]: You already loaded this key",
+            width=100,
+        )
+        return False
+
+    if not current_root.is_keyid_used(keyid):
+        console.print(
+            (
+                ":cross_mark: [red]Failed[/]: This key has not been used "
+                "to sign current root metadata",
+            ),
+            width=100,
+        )
+        return False
+
+    return True
+
+
+def _current_root_keys_validation(current_root: RootInfo):
+    """
+    Authorize user by loading current root threshold number of root keys
+    used for signing the current root metadata.
+    """
+    console.print(markdown.Markdown(AUTHORIZATION), width=100)
+    threshold = current_root.threshold
+    console.print(f"You will need to load {threshold} key(s).")
+    loaded: List[str] = []
+    key_count = 0
+    while key_count < current_root.threshold:
+        console.print(
+            f"You will enter information for key {key_count} of {threshold}"
+        )
+        root_key: RSTUFKey = _get_key(Root.type)
+        if root_key.error:
+            console.print(root_key.error)
+            try_again = prompt.Confirm.ask("Try again?")
+            if try_again:
+                continue
+            else:
+                raise click.ClickException(
+                    f"Failed authorization. Required: {threshold}, loaded "
+                    f"{len(loaded)} keys"
+                )
+
+        keyid = root_key.key["keyid"]
+        if not _is_valid_current_key(keyid, current_root, loaded):
+            continue
+
+        key_count += 1
+        loaded.append(keyid)
+        current_root.save_current_root_key(root_key)
+        console.print(
+            ":white_check_mark: Key "
+            f"{key_count}/{threshold} [green]Verified[/]"
+        )
+
+    console.print("\n[green]Authorization is successful [/]\n", width=100)
+
+
+def _keys_removal(current_root: RootInfo):
+    """Asking the user if he wants to remove any of the root keys"""
+    while True:
+        keys_table = _create_keys_table(current_root.keys, current_root, False)
+        console.print("Here are the current root keys:")
+        console.print(keys_table)
+        console.print("\n")
+
+        key_removal = prompt.Confirm.ask("Do you want to remove a key")
+        if not key_removal:
+            break
+
+        name = prompt.Prompt.ask(
+            "[green]Name/Tag/ID prefix[/] of the key to remove"
+        )
+        if not current_root.remove_key(name):
+            console.print(
+                "\n", f":cross_mark: [red]Failed[/]: key {name} is not in root"
+            )
+            continue
+        else:
+            console.print(f"Key with name/tag [yellow]{name}[/] removed\n")
+            if len(current_root.keys) < 1:
```

Comment:
IMO this check should be before you ask the user, if they want to remove a key. Otherwise, you could again have a situation, where you give a choice (want to remove y/n?) and then inform the user that they don't actually have a choice (remaining keys).

Maybe this is not a problem, because the loop cannot be entered with metadata that has no keys. But it seems easy to change this loop so that it does not require that precondition.